### ISSUE TITLE
Accessibility issues / Semantic HTML

### ISSUE THREAD

**Issue by Aeledfyr** [modrinth/code#2101]
The redesign is much better for this than the original site, but it still has some issues.  I'm mainly using Firefox's accessibility view to find issues, but most tooling should work.

Some of the issues:
In the header:
- [x] The modrinth logo is missing a label; since it's an `<svg>` element, it could use `aria-label="modrinth"`
- [x] The header could be a `<header>` element instead of `<aside>`
- [ ] The popup in the top right isn't labeled

On the search page:
- [x] Each mod should be contained in a `<section>` or `<article>` element instead of a `<div>`, so that each mod is separated semantically
- [x] The left and right pagination buttons and the pagination separator don't have labels.  `aria-label` could work (maybe even `title`?)  
- [x] The ui icons within each search result should have `aria-hidden="true"`, since they all have separate text labels
- [ ] The tooltips don't seem like they would be good for accessibility.  Is there a way of making that information available to screen readers and the like while keeping the same tooltip style?
    - Using `title` as a substitute could work, but it wouldn't look as nice.
- [ ] The filter selector isn't accessible since it uses `<p>` tags with click events; these could be replaced with buttons
- [ ] The dropdowns aren't great, but those aren't easy to fix or replace

On mod pages:
- [ ] The contrast for links in a mods description isn't great, but it should be fine
- [ ] The sidebar has the same issues as the search results for the ~~ui icons~~ and tooltips
- [ ] In featured versions, the download icon and fabric icon should have labels (`aria-label`, since they're svgs)
